contracts/src/Consensus.sol

```diff
@@ -0,0 +1,460 @@
+         * If valid, record attestation from all threshold participants?
+           OR record that the threshold was met for this tx?
+
+    ============================================================
+    EXTERNAL FUNCTIONS - THRESHOLD MANAGEMENT (OWNER ONLY)
+    ============================================================
+
+    8. proposeThresholdChange(uint128 newThreshold) external onlyOwner
+       - Propose a change to the threshold value
+       - Uses time-lock mechanism like Staking contract
+
+       Parameters:
+       - newThreshold: New threshold value
+
+       Validations:
+       - newThreshold > 1
+       - newThreshold <= current registered validator count - @CHECK Or should this be checked at execution? Or not checked at all as Validator count can change based on de/registrations)
+
+       Actions:
+       - Calculate executableAt = block.timestamp + CONFIG_TIME_DELAY
+       - Set pendingThresholdChange = ConfigProposal(newThreshold, executableAt)
+       - Emit ThresholdProposed(threshold, newThreshold, executableAt)
+
+    9. executeThresholdChange() external
+       - Execute a pending threshold change after timelock expires
+       - Can be called by anyone
+       - Change takes effect in the NEXT KeyGen, not current epoch
+
+       Validations:
+       - pendingThresholdChange.executableAt != 0 (proposal exists)
+       - block.timestamp >= pendingThresholdChange.executableAt
+       - pendingThresholdChange.value > 1
+       - pendingThresholdChange.value <= validatorsList.length (check at execution time)
+
+       Actions:
+       - Store old value: oldThreshold = threshold
+       - Update: threshold = pendingThresholdChange.value
+       - Clear proposal: delete pendingThresholdChange
+       - Emit ThresholdUpdated(oldThreshold, threshold)
+
+    ============================================================
+    VIEW FUNCTIONS
+    ============================================================
+
+    10. canIncrementEpoch() external view returns (bool)
+        - Returns true if enough time has passed to increment epoch
+        - Returns block.timestamp >= lastEpochTimestamp + 86400
+
+    11. getNextEpochTimestamp() external view returns (uint256)
+        - Returns timestamp when next epoch can start
+        - Returns lastEpochTimestamp + 86400
+
+    ============================================================
+    ADDITIONAL NOTES & FUTURE IMPROVEMENTS
+    ============================================================
+
+    1. Emergency Controls:
+       - @CHECK Should we be able to pause transaction proposals during security incidents
+       - Functions to pause: proposeSafeTransaction, attestTransaction
+
+    2. Transaction Lifecycle:
+       - Current spec: Transactions are proposed and attested, but never expire
+       - @CHECK Should we add transaction expiry (e.g., 30 days or 10 epochs)
+       - Add status: Pending, Attested (threshold met), Expired
```

Comment:
Similar question: should we have a "denied" state? Or would this be just a tx that expired?